handle triangular contour exceptions

zovelsanj opened the pull request:
In case any two coordinates of a contour are same, it will make a triangular region instead of rectangular one. Thus, when creating polygon from that contour, the area of the box (and thus distance) will be zero, resulting in an empty box. This results the following error during minimum area calculation:
```
bounding_box = cv2.minAreaRect(contour)
cv2.error: OpenCV(4.6.0) D:\a\opencv-python\opencv-python\opencv\modules\imgproc\src\convhull.cpp:143: error: (-215:Assertion failed) total >= 0 && (depth == CV_32F || depth == CV_32S) in function 'cv::convexHull'
```

### PR 类型 PR types
<!-- One of [ New features | Bug fixes | Function optimization | Performance optimization | Breaking changes | Others ] -->

### PR 变化内容类型 PR changes
<!-- One of [ Models | APIs | Docs | Others ] -->

### 描述 Description
<!-- Describe what this PR does -->

### 提PR之前的检查 Check-list

- [ ] 这个 PR 是提交到dygraph分支或者是一个cherry-pick，否则请先提交到dygarph分支。
      This PR is pushed to the dygraph branch or cherry-picked from the dygraph branch. Otherwise, please push your changes to the dygraph branch.
- [ ] 这个PR清楚描述了功能，帮助评审能提升效率。This PR have fully described what it does such that reviewers can speedup.
- [ ] 这个PR已经经过本地测试。This PR can be covered by existing tests or locally verified.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/PaddlePaddle/PaddleOCR?pullRequest=11887) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/PaddlePaddle/PaddleOCR?pullRequest=11887) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/PaddlePaddle/PaddleOCR?pullRequest=11887) it.</sub>

jzhang533 commented:
I think the fix is correct, but we changed the dev workflow, this patch should be applied to main branch.

zovelsanj commented:
> I think the fix is correct, but we changed the dev workflow, this patch should be applied to main branch.

Thanks @jzhang533, I have now changed the base to main.
